backend/tests/e2e/test_api_e2e.py

```
@@ -379,6 +376,227 @@
+            
+            # Validate that chronological_interactions combines both types correctly
+            llm_count_in_chrono = sum(1 for i in chronological_interactions if i.get("type") == "llm")
+            mcp_count_in_chrono = sum(1 for i in chronological_interactions if i.get("type") == "mcp")
+            
+            assert llm_count_in_chrono == len(llm_timestamps), \
+                f"LLM count mismatch in chronological_interactions: expected {len(llm_timestamps)}, got {llm_count_in_chrono}"
+            assert mcp_count_in_chrono == len(mcp_timestamps), \
+                f"MCP count mismatch in chronological_interactions: expected {len(mcp_timestamps)}, got {mcp_count_in_chrono}"
+            
+            # Deep validate ALL interactions in chronological_interactions list
+            for chrono_index, chrono_interaction in enumerate(chronological_interactions):
```

Ruff (0.12.2)

Line too long (90 > 88)

(E501)